src/rgw/rgw_op.cc

```diff
@@ -2681,6 +2681,35 @@ void RGWDeleteBucket::execute()
     }
   }
 
+  vector<rgw_bucket_dir_entry> objs;
+  string prefix = "", marker = "", delimiter = "";
+  map<string, bool> common_prefixes;
+  bool is_truncated;
+  static MultipartMetaFilter mp_filter;
+  int max_uploads = 1000;
```

Comment:
Make it a configurable or scavenge the whole bucket for incomplete uploads. Don't know exactly if 1000 would be sufficient for a heavy user who aborts uploads or a flaky network environment, if doesn't have s3 has restriction on that.